src/reducers/select.js

```diff
@@ -0,0 +1,222 @@
+    // (continue to normalization and comparison of contents --> contents are equal --> return base)
+    // Test 04: Base and source tags are not equal --> return base
+    // Note: to check this, the pattern has to allow two tags, otherwise the unequal tag does not even pass source.get(pattern)
+    if (baseField.tag === sourceField.tag === false) {
+      debug(`Tags are not equal`);
+      return base;
+    }
+    const baseSubs = baseField.subfields;
+    const sourceSubs = sourceField.subfields;
+
+    // Use .map, destructure code and value, and return a new object which has the same code and a new value for value
+    // Test 05: Normalize subfield values (base and source are equal) --> return base
+    const baseSubsNormalized = baseSubs.map(({code, value}) => ({code, value: normalizeSubfieldValue(value)}));
+    debug(`baseSubsNormalized: ${JSON.stringify(baseSubsNormalized, undefined, 2)}`);
+
+    const sourceSubsNormalized = sourceSubs.map(({code, value}) => ({code, value: normalizeSubfieldValue(value)}));
+    debug(`sourceSubsNormalized: ${JSON.stringify(sourceSubsNormalized, undefined, 2)}`);
+
+    // Compare equality of normalized subfields
+    // Test 06: Two subfields, both equal --> return base
+    // Test 07: Two subfields, same codes, values of source are subsets of values of base --> return source
+    // Test 08: Two subfields, one is a subset, one has a different code --> return base
+    // Test 09: Two subfields, same codes, values are not subsets --> return base
+    // Test 10: sourceField is a proper superset of baseField (subfields a and b are equal, c is new) --> return source
+    // Test 11: sourceField is not a proper superset of baseField (different values in a and b, also new subfield c) --> return base
+    // Test 12: sourceField is a proper superset of baseField (base values a and b are subsets of source values a and b, c is new in source) --> return source
+    // Test 13: Opposite of test 12, baseField is a proper superset of sourceField --> return base
+
+    // Filter out equal subfields (exactly equal code and value)
+    const equalSubfields = baseSubsNormalized.filter(isStrictEqualityTrue);
+    debug(`equalSubfields: ${JSON.stringify(equalSubfields, undefined, 2)}`);
+
+    // Subset comparison
+    const equalSubsets = baseSubsNormalized.filter(isSubsetEqualityTrue);
+    debug(`equalSubsets: ${JSON.stringify(equalSubsets, undefined, 2)}`);
+
+    // Run the equality functions in this order of priority
+    // How can this be done in a functional way?
+    // This way it does not take the return values (base or source) from the functions
+    // but goes all the way down to "Return original base" in every case
+    strictEquality(baseSubsNormalized, sourceSubsNormalized);
+    subsetEquality(baseSubsNormalized, sourceSubsNormalized);
+    strictEqualitySuperset(baseSubsNormalized, sourceSubsNormalized);
+    subsetEqualitySuperset(baseSubsNormalized, sourceSubsNormalized);
```

Comment:
You don't do this. You use `equalityFunction` which is declared in the top-level `export default` function. This function might be `strictEquality` or `subEquality` or some custom function. It doesn't matter. The function will return a boolean denoting whether the subfields are considered equal or not.

Using this function you can construct a list of subfields for which there is an equal counterpart in the opposite field.